packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -800,34 +800,48 @@ class FadeForwardsPageTransitionsBuilder extends PageTransitionsBuilder {
   ) => DualTransitionBuilder(
     animation: ReverseAnimation(secondaryAnimation),
     forwardBuilder: (BuildContext context, Animation<double> animation, Widget? child) {
-      return ColoredBox(
-        color:
-            animation.isAnimating
-                ? backgroundColor ?? Theme.of(context).colorScheme.surface
-                : Colors.transparent,
-        child: FadeTransition(
-          opacity: _fadeInTransition.animate(animation),
-          child: SlideTransition(
-            position: _secondaryForwardTranslationTween.animate(animation),
-            child: child,
-          ),
+      final Widget builder = FadeTransition(
+        opacity: _fadeInTransition.animate(animation),
+        child: SlideTransition(
+          position: _secondaryForwardTranslationTween.animate(animation),
+          child: child,
         ),
       );
+
+      final ModalRoute<dynamic>? route = ModalRoute.of(context);
+      if (route?.opaque ?? true) {
```

Comment:
We should be using an aspect of the ModalRoute here. However, I see that `opaque` is not an aspect of the modal route. (for example `isCurrent` is an aspect, accessible through `isCurrentOf()`)
I'll file a bug for that.

Once that is fixed we can do `ModalRoute.maybeIsOpaqueOf(context) ?? false` in this PR.

Reply:
Filed https://github.com/flutter/flutter/issues/167058 to follow up on that